### ISSUE TITLE
N802 false positives for `CGIHTTPRequestHandler` and `SimpleHTTPRequestHandler`

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#19422]
### Summary

#18907 added an exception to [`invalid-function-name` (N802)](https://docs.astral.sh/ruff/rules/invalid-function-name/) for `http.server.BaseHTTPRequestHandler`. That exception should also apply to its subclasses, `http.server.CGIHTTPRequestHandler` and `http.server.SimpleHTTPRequestHandler`. [Example](https://play.ruff.rs/5f94732e-948d-4404-9530-4e7ffffe8706):
```console
$ cat >n802.py <<'# EOF'
from http.server import CGIHTTPRequestHandler, SimpleHTTPRequestHandler
class C(CGIHTTPRequestHandler):
    def do_OPTIONS(self):
        pass
class S(SimpleHTTPRequestHandler):
    def do_OPTIONS(self):
        pass
# EOF

$ ruff --isolated check n802.py --select N802 --output-format concise -q
n802.py:3:9: N802 Function name `do_OPTIONS` should be lowercase
n802.py:6:9: N802 Function name `do_OPTIONS` should be lowercase
```

### Version

ruff 0.12.4 (ee2759b36 2025-07-17)
